src/provider/provider_trace.c

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) 2023 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "provider.h"
+
+#include <umf/memory_provider_ops.h>
+
+#include <assert.h>
+#include <stdlib.h>
+
+struct traceParams {
```

Comment:
there should be 2 types of trace structure - params and priv. Params should be put in headers, and priv should be defined here. Same for the pools

Reply:
Done